crates/orchestrator/src/tests/workers/mod.rs

```diff
@@ -0,0 +1,85 @@
+use crate::config::config_force_init;
+use crate::database::MockDatabase;
+use crate::jobs::types::{ExternalId, JobItem, JobStatus, JobType};
+use crate::queue::job_queue::JobQueueMessage;
+use crate::queue::MockQueueProvider;
+use crate::tests::common::init_config;
+use crate::workers::snos::SnosWorker;
+use crate::workers::Worker;
+use da_client_interface::MockDaClient;
+use httpmock::MockServer;
+use rstest::rstest;
+use serde_json::json;
+use std::collections::HashMap;
+use uuid::Uuid;
+
+#[rstest]
+#[tokio::test]
+async fn test_create_job() {
+    let server = MockServer::start();
+    let da_client = MockDaClient::new();
+    let mut db = MockDatabase::new();
+    let mut queue = MockQueueProvider::new();
+
+    // Mocking db functions
+    db.expect_get_latest_job_by_type().returning(|_| Ok(None)).call(JobType::SnosRun).expect("Failed to call.");
+    // Getting jobs for check expectations
+    for i in 1..6 {
+        db.expect_get_job_by_internal_id_and_type()
+            .returning(|_, _| Ok(None))
+            .call(&i.to_string(), &JobType::SnosRun)
+            .expect("Failed to call.");
+    }
+
+    // Creating jobs expectations
+    for i in 1..6 {
+        db.expect_create_job()
+            .returning(|_| Ok(get_job_item_mock_by_id("1".to_string())))
+            .call(get_job_item_mock_by_id(i.to_string()))
```

Comment:
should we just call `get_job_item_mock_by_id` once and use it in both places?

Reply:
doesn't work tried this value move ho jaati h

Reply:
try cloning?

Reply:
already tried

Reply:
Cloning a string should ideally work, what were you doing exactly?